encoding/graph/thing/vertex_generator.rs

```diff
@@ -326,12 +331,13 @@ impl ThingVertexGenerator {
     where
         Snapshot: WritableSnapshot,
     {
-        if string.length() <= StringAttributeID::ENCODING_INLINE_CAPACITY {
+        if StringAttributeID::is_inlineable(string.as_reference()) {
             Ok(StringAttributeID::build_inline_id(string))
         } else {
             let id = StringAttributeID::build_hashed_id(type_id, string, snapshot, &self.large_value_hasher)?;
             let hash = id.get_hash_prefix_hash();
-            let lock = ByteArray::copy_concat([&type_id.bytes(), &hash]);
+            let lock =
+                ByteArray::copy_concat([&Prefix::VertexAttributeString.prefix_id().bytes(), &type_id.bytes(), &hash]);
```

Comment:
Why not just lock the whole ID?

Reply:
HMMM I think I have done this because two concurrent transactions can end up hashing multiple values (by pure bad luck) into the same hash value, which will lead transaction 1 to use disambiguation byte = `0`, and the other one to use disambiguation byte `= 1` and we won't detect the conflict... Like this we force things that collide on the hash to fail (which should be extremely rarely anyway).